define/types.go

```diff
@@ -101,11 +101,43 @@ type Secret struct {
 
 // BuildOutputOptions contains the the outcome of parsing the value of a build --output flag
 type BuildOutputOption struct {
+	Type     BuildOutputType
+	Attrs    map[string]string
 	Path     string // Only valid if !IsStdout
 	IsDir    bool
 	IsStdout bool
 }
 
+type BuildOutputType int
+
+const (
```

Comment:
Consider naming the constants `BuildOutput[Type]…`, to make the relationship clearer. (Another price we pay for Go’s “simplicity”.)